tests/Phinx/Db/Adapter/MysqlAdapterTest.php

```diff
@@ -890,6 +892,102 @@ public function testTinyTextColumn()
         $this->assertEquals('tinytext', $sqlType['name']);
     }
 
+    public function binaryToBlobAutomaticConversionData()
+    {
+        return [
+          [null, 'binary', 255],
+          [64, 'binary', 64],
+          [MysqlAdapter::BLOB_REGULAR - 20, 'blob', MysqlAdapter::BLOB_REGULAR],
+          [MysqlAdapter::BLOB_REGULAR, 'blob', MysqlAdapter::BLOB_REGULAR],
+          [MysqlAdapter::BLOB_REGULAR + 20, 'mediumblob', MysqlAdapter::BLOB_MEDIUM],
+          [MysqlAdapter::BLOB_MEDIUM, 'mediumblob', MysqlAdapter::BLOB_MEDIUM],
+          [MysqlAdapter::BLOB_MEDIUM + 20, 'longblob', MysqlAdapter::BLOB_LONG],
+          [MysqlAdapter::BLOB_LONG, 'longblob', MysqlAdapter::BLOB_LONG],
+          [MysqlAdapter::BLOB_LONG + 20, 'longblob', MysqlAdapter::BLOB_LONG],
+        ];
+    }
+
+    /** @dataProvider binaryToBlobAutomaticConversionData */
+    public function testBinaryToBlobAutomaticConversion(?int $limit = null, string $expectedType, int $expectedLimit)
+    {
+        $table = new \Phinx\Db\Table('t', [], $this->adapter);
+        $table->addColumn('column1', 'binary', ['limit' => $limit])
+              ->save();
+        $columns = $table->getColumns();
+        $sqlType = $this->adapter->getSqlType($columns[1]->getType(), $columns[1]->getLimit());
+        $this->assertEquals($expectedType, $sqlType['name']);
+        $this->assertEquals($expectedLimit, $columns[1]->getLimit());
+    }
+
+    public function varbinaryToBlobAutomaticConversionData()
+    {
+        return [
+          [null, 'varbinary', 255],
+          [64, 'varbinary', 64],
+          [MysqlAdapter::BLOB_REGULAR - 20, 'blob', MysqlAdapter::BLOB_REGULAR],
+          [MysqlAdapter::BLOB_REGULAR, 'blob', MysqlAdapter::BLOB_REGULAR],
+          [MysqlAdapter::BLOB_REGULAR + 20, 'mediumblob', MysqlAdapter::BLOB_MEDIUM],
+          [MysqlAdapter::BLOB_MEDIUM, 'mediumblob', MysqlAdapter::BLOB_MEDIUM],
+          [MysqlAdapter::BLOB_MEDIUM + 20, 'longblob', MysqlAdapter::BLOB_LONG],
+          [MysqlAdapter::BLOB_LONG, 'longblob', MysqlAdapter::BLOB_LONG],
+          [MysqlAdapter::BLOB_LONG + 20, 'longblob', MysqlAdapter::BLOB_LONG],
+        ];
+    }
+
+    /** @dataProvider varbinaryToBlobAutomaticConversionData */
+    public function testVarbinaryToBlobAutomaticConversion(?int $limit = null, string $expectedType, int $expectedLimit)
+    {
+        $table = new \Phinx\Db\Table('t', [], $this->adapter);
+        $table->addColumn('column1', 'varbinary', ['limit' => $limit])
+              ->save();
+        $columns = $table->getColumns();
+        $sqlType = $this->adapter->getSqlType($columns[1]->getType(), $columns[1]->getLimit());
+        $this->assertEquals($expectedType, $sqlType['name']);
+        $this->assertEquals($expectedLimit, $columns[1]->getLimit());
```

Comment:
Can we please use assertSame() everywhere to make sure the types are also as expected?

Reply:
Sure. Should I also switch all others assertEquals ? I've simply mirrored pattern found in test file.

Reply:
If you want, sure.
We can also do that in a separate PR for other code, though.